python/ray/rllib/policy_gradient/agent.py

```diff
@@ -148,11 +150,44 @@ def load_weights(self, weights):
         self.variables.set_weights(weights)
 
     def compute_trajectory(self, gamma, lam, horizon):
+        """Compute a single rollout on the agent and return."""
         trajectory = rollouts(
             self.common_policy,
             self.env, horizon, self.observation_filter, self.reward_filter)
         add_advantage_values(trajectory, gamma, lam, self.reward_filter)
         return trajectory
 
+    def compute_steps(self, gamma, lam, horizon, min_steps_per_task=-1):
+        """Compute multiple rollouts and concatenate the results.
+
+        Parameters:
+            num_steps: Lower bound on the number of states to be collected.
+        Returns:
+            states: List of states.
+            total_rewards: Total rewards of the trajectories.
+            traj_lengths: Lengths of the trajectories.
+        """
+        num_steps_so_far = 0
+        trajectories = []
+        total_rewards = []
+        traj_lengths = []
+        while True:
+            trajectory = rollouts(
+                self.common_policy,
+                self.env, horizon, self.observation_filter, self.reward_filter)
+            add_advantage_values(trajectory, gamma, lam, self.reward_filter)
+            total_rewards.append(
+                trajectory["raw_rewards"].sum(axis=0).mean())
+            traj_lengths.append(np.logical_not(trajectory["dones"]).sum(axis=0).mean())
+            trajectory = flatten(trajectory)
+            not_done = np.logical_not(trajectory["dones"])
+            trajectory = {key: val[not_done]
```

Comment:
add a comment saying that we're filtering out the environments that are finished